File: Wayne/WayneCompiler/src/Dfir_representation/ICaseSelector.java
package Dfir_representation;

import java.util.Vector;

public class ICaseSelector extends IBorderNode{
	   private int defaultDiagramIndex;
	   private Vector<Range> ranges;
	   public ICaseSelector(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals, 
			                int defaultDiagramIndex,Vector<Range> ranges){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   this.JNodeType = "ICaseSelector";
		   this.defaultDiagramIndex = defaultDiagramIndex;
		   this.ranges = ranges;
	   }
	    public int getDefaultDiagramIndex() {
			return defaultDiagramIndex;
		}

		public void setDefaultDiagramIndex(int defaultDiagramIndex) {
			this.defaultDiagramIndex = defaultDiagramIndex;
		}

		public Vector<Range> getRanges() {
			return ranges;
		}

		public void setRanges(Vector<Range> ranges) {
			this.ranges = ranges;
		}
		
		public void decIndegree(){
			   this.indegree--;
			   JNode.IdToJNode.get(this.parentId).indegree--;			   
		   }
	  
	   public void print(){
		   System.out.println("ICaseSelector begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
		   System.out.println("DefaultDiagramIndex: " + this.defaultDiagramIndex);
		   for(int i=0; i<this.ranges.size(); i++){
			   this.ranges.get(i).print();
		   }
		   System.out.println("ICaseSelector end\n");
	   }
	   public Boolean getIsConnectedForIDiagram(IDiagram iDiagram){
		   Boolean res = false;
		   if(this.getOutputTerminals().get(iDiagram.getDiagramIndex()).getConnections().size()!=0){
			   res = true;
		   }
		   return res;
	   }
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/ICaseSelector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DFIR_Compiler
{
    public class ICaseSelector:IBorderNode
    {
        private int defaultDiagramIndex;
        private List<Range> ranges;

        public ICaseSelector(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals, int DefaultDiagramIndex, List<Range> Ranges) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            INodeType = "ICaseSelector";
            defaultDiagramIndex = DefaultDiagramIndex;
            ranges = Ranges;
        }


        public virtual int GetDefaultDiagramIndex()
        {
            return defaultDiagramIndex;
        }
        public virtual List<Range> GetRanges()
        {
            return ranges;
        }


        public virtual void SetDefaultDiagramIndex(int DefaultDiagramIndex)
        {
            defaultDiagramIndex = DefaultDiagramIndex;
        }
        public virtual void SetRanges(List<Range> Ranges)
        {
            ranges = Ranges;
        }


        public virtual bool IsConnectedIDiagram(IDiagram iDiagram)
        {
            bool flag = false;
            if (GetOutputTerminals()[iDiagram.getDiagramIndex()].GetConnections().Count != 0)
            {
                flag = true;
            }
            return flag;
        }


        public override void DecreaseIndegree()
        {
            indegree--;
            IdToINode[parentId].indegree--;
        }


        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("ICaseSelector:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
            Console.WriteLine("DefaultDiagramIndex: " + defaultDiagramIndex);
            for (int i = 0; i < ranges.Count; i++)
            {
                ranges[i].Print();
            }
        }


    }
}
